python/paddle/fluid/tests/unittests/test_set_value_op.py

```diff
@@ -915,7 +917,315 @@ def test_dynamic(self):
+        self.assertTrue(
+            np.array_equal(inps.grad.numpy(), input_grad),
+            msg="The gradient of value should be \n{},\n but reveived {}".
+            format(input_grad, inps.grad.numpy()))
+        self.assertTrue(
+            np.array_equal(value.grad.numpy(), value_grad),
+            msg="The gradient of input should be \n{},\n but reveived {}".
+            format(value_grad, value.grad.numpy()))
+
+        # case 5:a[0].shape==value.shape
+        def set_value5(t, value):
+            a = t * t
+            a[0] = value
+            y = a * a
+            return y.sum()
+
+        array = np.arange(1, 1 + 2 * 3 * 4, dtype="float32").reshape([2, 3, 4])
+        value = np.arange(100, 100 + 12, dtype="float32").reshape(3, 4)
+
+        inps = paddle.to_tensor(array, stop_gradient=False)
+        value = paddle.to_tensor(value, stop_gradient=False)
+
+        loss = set_value5(inps, value)
+        loss.backward()
+
+        value_grad = np.array([[200., 202., 204., 206.],
+                               [208., 210., 212., 214.],
+                               [216., 218., 220., 222.]])
+        input_grad = np.array([[[0., 0., 0., 0.], [0., 0., 0., 0.],
+                                [0., 0., 0., 0.]],
+                               [[8788., 10976., 13500., 16384.],
+                                [19652., 23328., 27436., 32000.],
+                                [37044., 42592., 48668., 55296.]]])
+        self.assertTrue(
+            np.array_equal(inps.grad.numpy(), input_grad),
+            msg="The gradient of value should be \n{},\n but reveived {}".
+            format(input_grad, inps.grad.numpy()))
+        self.assertTrue(
+            np.array_equal(value.grad.numpy(), value_grad),
+            msg="The gradient of input should be \n{},\n but reveived {}".
+            format(value_grad, value.grad.numpy()))
+
+    def test_static_graph(self):
+        paddle.enable_static()
+
+        to_string = lambda x, i, : x + '_' + str(i)
+        numel = lambda input_shape: reduce(lambda x, y: x * y, input_shape)
+
+        def op1(x):
+            # value.stop_gradient = True
+            # x.stop_gradient = False
+            value = paddle.fluid.layers.fill_constant([1], "float32", 1)
+            start = paddle.fluid.layers.fill_constant(
+                [1], "int32", 5, force_cpu=True)
+            end = paddle.fluid.layers.fill_constant(
+                [1], "int32", 0, force_cpu=True)
+            step = paddle.fluid.layers.fill_constant(
+                [1], "int32", -2, force_cpu=True)
+
+            inputs = {
+                'Input': x,
+                'ValueTensor': value,
+                'StartsTensorList': [start, ],
+                'EndsTensorList': [end, ],
+                'StepsTensorList': [step, ]
+            }
+
+            helper = LayerHelper("set_value")
+            y = helper.create_variable_for_type_inference(dtype=x.dtype)
+
+            helper.append_op(
```

Comment:
why need this form to test? form like `a[0] = value` can't work in static mode?

Reply:
set_value_op in __get/setitem__ is an inplace operation. When `input.stop_gradient = True` and `value.stop_gradient = False`,  set_value_grad_op will not be run during backward.